packages/@lwc/synthetic-shadow/src/polyfills/mouse-event/polyfill.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { defineProperty, isNull, getOwnPropertyDescriptor } from '@lwc/shared';
+import { pathComposer } from '../../3rdparty/polymer/path-composer';
+import { retarget } from '../../3rdparty/polymer/retarget';
+
+const mouseEventRelatedTargetGetter = getOwnPropertyDescriptor(
+    FocusEvent.prototype,
+    'relatedTarget'
+)!.get as () => EventTarget | null;
+
+defineProperty(MouseEvent.prototype, 'relatedTarget', {
+    get(this: Event): EventTarget | null | undefined {
+        const currentTarget = this.currentTarget;
```

Comment:
```suggestion
        const currentTarget = eventCurrentTargetGetter.call(this);
```

Reply:
intentionally using `this.currentTarget` here since it is used below for the retargeting logic. not sure what the implication is if we change it to the original current target.